wasmtime-runtime/src/jit_dump.rs

```diff
@@ -0,0 +1,650 @@
+//! Support for jitdump files which can be used by perf for profiling jitted code.
+//! https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/perf/Documentation/jitdump-specification.txt
+//!
+//! Usage Example:
+//!     Record
+//!         sudo perf record -k 1 -e instructions:u target/debug/wasmtime -g -jitdump test.wasm
+//!     Combine
+//!         sudo perf inject -v -j -i perf.data -o perf.jit.data
+//!     Report
+//!         sudo perf report -i perf.jit.data -F+period,srcline
+//! Note: For descriptive results, the WASM file being executed should contain dwarf debug data
+
+use libc::{c_int, c_void, clockid_t, mmap, open, sysconf, timespec};
+use serde::{Deserialize, Serialize};
+use std::ffi::CString;
+use std::fmt::Debug;
+use std::fs::File;
+use std::io::Write;
+use std::{mem, process, borrow};
+use std::os::unix::io::FromRawFd;
+use object::Object;
+use target_lexicon::Architecture;
+
+#[cfg(target_pointer_width = "64")]
+use goblin::elf64 as elf;
+
+#[cfg(target_pointer_width = "32")]
+use goblin::elf32 as elf;
+
+#[repr(C)]
```

Comment:
```suggestion
#[repr(u32)]
```

Reply:
Thanks for the suggestion. Can you explain a little more thinking behind this change? I see repr(C) used throughout wasmtime (for structs at least) and the rust reference points out using repr(u*) for fieldless enums, which this is not. Why do you prefer repr(u32) over repr(C) in this case?

Reply:
Because you cast it to a `u32` at several places. Also you don't seem to actually use the `#[repr(C)]`ness.

Reply:
Ok .. the change is no problem, I just want to understand the impact of the update. I am having a hard time finding any details about repr(u32) to the level of repr(C). I understand repr(C) to guarantee the sizes and ordering of the types are exactly what I'd expect in C which is what I want. For repr(u*) .. the docs I'm reading only mention fieldless enums and don't mention any details about how this attribute affects type sizes or ordering of struct elements. What does repr(u32) define differently from repr(C)?

Reply:
repr(u32) is guaranteed to have a 32bit size in all cases. It will error when that is not possible.

https://doc.rust-lang.org/stable/reference/type-layout.html#primitive-representations

Reply:
Ahh ... I see. This is only relevant for the Enum. Ok, thanks.